src/components/products/product_details.jsx

```diff
@@ -0,0 +1,118 @@
+import React, { Component } from "react";
+import { connect } from "react-redux";
+import { clearProductDetails, getProductDetails, addItemToCart } from "../../action/index";
+import Money from "../general/money";
+import "bootstrap/dist/css/bootstrap.css";
+import "./products.scss";
+import Modal from "../general/modal";
+
+class ProductDetails extends Component {
+   constructor(props) {
+      super(props);
+   }
+   state = {
+      quantity: 1,
+   };
+
+   componentDidMount = () => {
+      if (this.props.match == undefined) return false;
+      const {
+         getProductDetails,
+         match: { params },
+      } = this.props;
+      getProductDetails(params.product_id);
+   };
+
+   componentWillMount = () => {
```

Comment:
`componentDidMount` is deprecated and should not be used anymore. This could be done in the `constructor` or in this case it is best to clear the product details once someone leave this page so `componentWillUnmount` would be best here.